Doc/library/stdtypes.rst

```diff
@@ -4759,46 +4759,47 @@ Generic Alias Type
-<classmethod>` :meth:`__class_getitem__` of the class instead. The classmethod
-:meth:`__class_getitem__` returns a ``GenericAlias`` object.
+Usually, the :ref:`subscription <subscriptions>` of builtin containers
+objects calls the method :meth:`__getitem__` of the object.  However, the
+subscription of some builtin containers' classes may call the classmethod
+:meth:`__class_getitem__` of the class instead. The classmethod
+:meth:`__class_getitem__` returns a ``GenericAlias`` object. An example of this
+is the expression ``list[int]``.
+
+.. note::
+   When :meth:`__class_getitem__` of the class is not present, the
+   :meth:`__getitem__` is still called.
 
 The ``GenericAlias`` object acts as a proxy for :term:`generic types
 <generic type>`, implementing *parameterized generics* - a specific instance
-of a generic with the types for container elements provided. It is intended
-primarily for type :term:`annotations <annotation>`.  ``GenericAlias`` objects
-are created by expressions like ``list[int]``.
+of a generic which provides the types for container elements.  It is intended
+primarily for :term:`type annotations <annotation>`.
 
-The type for the ``GenericAlias`` object is :data:`types.GenericAlias`.
-
-.. impl-detail::
+The user-exposed type for the ``GenericAlias`` object can be accessed from
+:data:`types.GenericAlias` and used for :func:`isinstance` checks.
 
-   The C equivalent of :meth:`__getitem__` - :c:func:`PyObject_GetItem` - is
-   called first for builtin classes and types. When :c:func:`PyObject_GetItem`
-   detects that it was called on a type, it then calls :meth:`__class_getitem__`
-   and returns the ``GenericAlias`` object.
+.. describe:: GenericType[X, Y, ...]
 
-.. describe:: generic[X, Y, ...]
-
-   Defines a generic containing elements of types *X*, *Y*, and more depending
-   on the *generic* used.  For example, for a function expecting a :class:`list`
-   containing :class:`float` elements::
+   Creates a ``GenericAlias`` representing a ``GenericType`` containing elements
+   of types *X*, *Y*, and more depending on the ``GenericType`` used.
+   For example, for a function expecting a :class:`list` containing
+   :class:`float` elements::
 
       def average(values: list[float]) -> float:
           return sum(values) / len(values)
 
-   Another example for :term:`mapping` objects, using a :class:`dict`.  In this
-   case, the function expects a ``dict`` for its ``body`` argument. The ``body``
-   has keys of type :class:`str` and their corresponding values are lists which
-   hold :class:`int` elements::
+   Another example for :term:`mapping` objects, using a :class:`dict`. A
+   :class:`dict` is a generic type expecting two type parameters representing
+   the key type and the value type.  In this case, the function expects a
+   ``dict`` for its ``body`` argument.  The ``body`` has keys of type
+   :class:`str` and their corresponding values are lists which hold :class:`int`
+   elements::
```

Comment:
Honestly the example would be easier to follow if you just use something like `dict[str, int]`. We don't need to educate people about nesting generic types through this example.